src/App.jsx

```diff
@@ -14,45 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPH_BUTTON_ID = 'alphabetically';
+const LENGTH_BUTTON_ID = 'length';
+
+function getPrepareGoods(goods, sortField, reverse) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case ALPH_BUTTON_ID:
+          return good1.localeCompare(good2);
+        case LENGTH_BUTTON_ID:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const visibleGoods = getPrepareGoods(goodsFromServer, sortField, isReverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const onReset = () => {
+    setSortField('');
+    setIsReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortField(ALPH_BUTTON_ID);
+          }}
+          type="button"
+          className={cn('button is-info',
+            sortField !== ALPH_BUTTON_ID && 'is-light')}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            setSortField(LENGTH_BUTTON_ID);
+          }}
+          type="button"
+          className={cn('button is-success',
+            sortField !== LENGTH_BUTTON_ID && 'is-light')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setIsReverse(!isReverse);
+          }}
+          type="button"
+          className={cn('button is-warning',
+            !isReverse && 'is-light')}
+        >
+          Reverse
+        </button>
+        {(sortField || isReverse) && (
```

Comment:
you can create separate variable like `isShowResetButton`
this is will be more readable
```suggestion
        const isShowResetButton = sortField || isReverse;
...
        {isShowResetButton && (
```